stellarphot/settings/tests/test_custom_widgets.py

```diff
@@ -816,44 +839,420 @@ def test_title_format(self):
+        # Check that if there are already settings saved to the working directory
+        # and the user clicks on the tab, the badge is updated to reflect that the
+        # setting has been saved.
+
+        # To set this up we need to save settings to the working directory AND to the
+        # saved user settings.
+        wd_settings = PhotometryWorkingDirSettings()
+        camera = Camera(**TEST_CAMERA_VALUES)
+        observatory = Observatory(**DEFAULT_OBSERVATORY_SETTINGS)
+        passbands = PassbandMap(**DEFAULT_PASSBAND_MAP)
+        wd_settings.save(
+            PartialPhotometrySettings(
+                camera=camera, observatory=observatory, passband_map=passbands
+            )
+        )
+        saved = SavedSettings()
+        saved.add_item(camera)
+        saved.add_item(observatory)
+        saved.add_item(passbands)
+
+        review_settings = ReviewSettings(
+            [
+                Camera,
+                Observatory,
+                PassbandMap,
+                PhotometryApertures,
+                SourceLocationSettings,
+                PhotometryOptionalSettings,
+                LoggingSettings,
+            ]
+        )
+
+        num_tabs = len(review_settings._container.children)
+
+        for title in review_settings._container.titles:
+            assert SaveStatus.SETTING_SHOULD_BE_REVIEWED in title
+
+        # Click on each tab, starting with the last one, to make sure a change in
+        # the selected value happens to trigger the observers.
+        for i in range(num_tabs - 1, -1, -1):
+            review_settings._container.selected_index = i
+
+        for title in review_settings._container.titles:
+            assert SaveStatus.SETTING_IS_SAVED in title
+
+    def test_loading_saved_source_locations_to_ui(self):
+        # Check that saved source locations are loaded into the UI
+        wd_set = PhotometryWorkingDirSettings()
+        source_locations = SourceLocationSettings()
+        wd_set.save(
+            PartialPhotometrySettings(source_location_settings=source_locations)
+        )
+
+        review_settings = ReviewSettings([SourceLocationSettings])
+
+        # Check that the saved source locations are in the UI
+        assert (
+            review_settings._setting_widgets[0]._autoui_widget.value
+            == source_locations.model_dump()
+        )
+        assert (
+            review_settings._setting_widgets[0]
+            ._autoui_widget.di_widgets["source_list_file"]
+            .selected_filename
+            == source_locations.source_list_file
+        )
+
+    def test_getting_settings_with_nothing_saved(self):
+        # Check that when we create the object with no saved settings we get an empty
+        # partial settings object.
+        review_settings = ReviewSettings([Camera])
+        assert review_settings.current_settings == PartialPhotometrySettings()
+
+    def test_selecting_table_without_saved_setting_sets_proper_badge(self):
```

Comment:
Maybe a comment to clarify the goal.  Maybe add
```
        # Test that selecting a tab with a saved setting sets a proper "NOT SAVED" badge
```

Reply:
done